apps/api/src/domain/models/resource.entity.ts

```diff
@@ -1,39 +1,60 @@
 import { PartialDTO } from 'apps/api/src/types/partial-dto';
+import { InternalError } from '../../lib/errors/InternalError';
+import { Valid } from '../domainModelValidators/Valid';
 import { EntityId } from '../types/ResourceId';
 import { ResourceType } from '../types/resourceTypes';
+import { getAllowedContextsForModel } from './allowedContexts/isContextAllowedForGivenResourceType';
 import BaseDomainModel from './BaseDomainModel';
+import { ContextModelUnion } from './context/types/ContextModelUnion';
 import { EdgeConnectionContextType } from './context/types/EdgeConnectionContextType';
 import { ResourceCompositeIdentifier } from './types/entityCompositeIdentifier';
 
 export abstract class Resource extends BaseDomainModel {
-    abstract readonly type: ResourceType;
+    readonly type: ResourceType;
 
     readonly id: EntityId;
 
-    abstract readonly allowedContextTypes: EdgeConnectionContextType[];
+    readonly allowedContextTypes: EdgeConnectionContextType[];
 
     // TODO: Rename this 'isPublished' - db migration
     readonly published: boolean;
 
     constructor(dto: PartialDTO<Resource>) {
         super();
 
+        this.type = dto.type;
+
         this.id = dto.id;
 
         this.published = typeof dto.published === 'boolean' ? dto.published : false;
+
+        this.allowedContextTypes = getAllowedContextsForModel(this.type);
     }
 
     getCompositeIdentifier = (): ResourceCompositeIdentifier => ({
         type: this.type,
         id: this.id,
     });
 
-    // protected registerAllowedContextType<TResource extends Resource>(
-    //     this: TResource,
-    //     contextType: EdgeConnectionContextType
-    // ): void {
-    //     this.clone<TResource>({
-    //         allowedContextTypes: [...new Set(this.allowedContextTypes.concat(contextType))],
-    //     } as Partial<TResource>);
-    // }
+    /**
+     * We choose to put the invariant validation in the factory so not to
```

Comment:
I prefer choose, because it indicates that it is always a state of being- we can change our minds at any times. If only English had an aspect for `We have chosen to and continue to choose to _____`, I'd use that.